src/llm/wmt_model.py

```diff
@@ -13,6 +13,18 @@ class TransformerConfig:
     expansion_dim: int
     num_heads: int
     num_blocks: int
+    dropout_rate: float
+
+
+class ResidualDropout(nn.Module):
```

Comment:
I think this is a redundant `nn.Dropout` Module.

It does not add the residual bit to the dropout bit.

I am open to keeping it, provided it does something along with applying dropout.

Reply:
nice catch. moved residual (add) inside this class.